client.go

```diff
@@ -728,6 +741,11 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 		// without racing against the closeBody call in persistConn.writeLoop.
 		httpreq := *req.Request
 		req.Request = &httpreq
+
+		if err := c.Postprocess(req.Request); err != nil {
+			checkErr = err
```

Comment:
I'm not sure if they will like reusing checkErr that is used by CheckRetry method

Reply:
Yes, it makes sense to add separate error variable for that